Spec terms of service at registration (MSC1692) (#1812)
matrix-org/matrix-spec: content/client-server-api/_index.md

{{% http-api spec="client-server" api="registration_tokens" %}}

##### Terms of service at registration

{{% added-in v="1.11" %}}

| Type                     | Description                                                           |
|--------------------------|-----------------------------------------------------------------------|

> | `m.login.terms`          | Authentication requires the user to accept a set of terms of service. |

Reviewer: ```suggestion
| `m.login.terms`          | Authentication requires the user to accept a set of terms of policy documents. |
```

to be consistent with [the paragraph below the note](https://github.com/matrix-org/matrix-spec/pull/1812/files#diff-c82b624572233237a687eb9baaf76eafd820da0b362057ed0851593343a37d3bR997)

Author: Applied a slightly different version.

> | Type                     | Description                                                           |
> |--------------------------|-----------------------------------------------------------------------|
> | `m.login.terms`          | Authentication requires the user to accept a set of terms of service. |

Author: ```suggestion
| Type                     | Description                                                              |
|--------------------------|--------------------------------------------------------------------------|
| `m.login.terms`          | Authentication requires the user to accept a set of policy documents. |
```

{{% boxes/note %}}
The `m.login.terms` authentication type is only valid on the
[`/register`](#post_matrixclientv3register) endpoint.
{{% /boxes/note %}}

This authentication type is used when the homeserver requires new users to

> accept a given set of policy douments, such as a terms of service and a privacy

Reviewer: ```suggestion
accept a given set of policy documents, such as a terms of service and a privacy
```

policy. There may be many different types of documents, all of which are
versioned and presented in (potentially) multiple languages.

When the server requires the user to accept some terms, it does so by returning
a 401 response to the `/register` request, where the response body includes
`m.login.terms` in the `flows` list, and the `m.login.terms` property in the
`params` object has the structure [shown below](#definition-mloginterms-params).

If a client encounters an invalid parameter, registration should stop with an
error presented to the user.

The client should present the user with a checkbox to accept each policy,

> including a link to the provided url. Once the user has done so, the client

Reviewer: ```suggestion
including a link to the provided URL. Once the user has done so, the client
```

Reviewer: or 
```suggestion
including a link to the provided `url`. Once the user has done so, the client
```

if you're referring to the property in the policy definition
